### PR TITLE
Adding WearTouchPad extension

### PR DESCRIPTION
**Pull request by parthmehta209**
WearTouchPad is a GearVRf extension. It is an app that should be installed on the phone.
This app has a companion Android wear app. The mobile app gets all the touchpad events from
the companion Android wear app. This extension is used by the GVRAndroidWearTouchpad for getting touchpad events from the Android wear watch and delivering it to the GVRActivity.
The GVRf/Extensions/WearTouchPad/WearTouchPad.apk apk file is a combined apk file that will install the app on the phone as well as the connected Android wear watch. This is included for easily installing the app on the watch and the phone.

### CODE REVIEW

**Comment by thomasflynn** [comment]
fwiw, it turns out i needed to upgrade to java 8 to compile.  now to actually try it...

**Comment by thomasflynn** [comment]
ok, good, i can confirm this is functional now.  +1, looks good to me.

File: GVRf/Framework/backend_oculus/src/main/java/org/gearvrf/OvrXMLParser.java [hunk 1]
```diff
@@ -106,6 +106,10 @@
                                 settings.setUseGazeCursorController(Boolean
                                         .parseBoolean(xpp.getAttributeValue(i)));
                             } else if (attributeName
+                                    .equals("useAndroidWearTouchpad")) {
```
Comment (liaxim) [inline GVRf/Framework/backend_oculus/src/main/java/org/gearvrf/OvrXMLParser.java]: You will need to update the parsers for the other backends to recognize the new setting. Clearly we need a common parser but for now they are separate.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRAndroidWearTouchpad.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
+package org.gearvrf.io;
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRAndroidWearTouchpad.java]: Needs copyright note.

**Comment by liaxim** [comment]
Do you want something integrated with the automated build? Any publishable build artifacts?

--

It is not really an extension in the usual terms. If it were a real extension (like a separate .aar) it would be the best if the feature gets automatically enabled as long as the aar is included in the app. No need for extra setting in the xml file. Should we try to make it a real extension? What do you think?

I know there will be a bit of work involved since we haven't had that type of extension before.

File: GVRf/Extensions/WearTouchPad/mobile/build.gradle [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
+    wearApp project(':wear')
```
Comment (liaxim) [inline GVRf/Extensions/WearTouchPad/mobile/build.gradle]: What does ``wearApp project(':wear')`` do?

File: GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRAndroidWearTouchpad.java (hunk 2 above)
Comment (parthmehta209) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRAndroidWearTouchpad.java]: Done.

File: GVRf/Framework/backend_oculus/src/main/java/org/gearvrf/OvrXMLParser.java (hunk 1 above)
Comment (parthmehta209) [inline GVRf/Framework/backend_oculus/src/main/java/org/gearvrf/OvrXMLParser.java]: Yes. I will make the change in other backends.

**Comment by parthmehta209** [comment]
@Liahim I am not sure how we can make an aar. The reason I made a separate app is that each app that has a companion wear app has to have the same application ID as the wear app. As far as I know we cannot have a wearable app inside an aar. But I can investigate further into making it more like a regular extension. I have added the .apk file in this checkin. So I we would not need a build artifact for this extension. Or we can have the automated build create a signed apk and have it as one of the artifacts uploaded to github.

**Comment by liaxim** [comment]
@parthmehta209 and I had a chat. We are not going to make big changes for this pull request. Will look into the extension thing in the future.

File: GVRf/Framework/backend_oculus/src/main/java/org/gearvrf/OvrXMLParser.java (hunk 1 above)
Comment (parthmehta209) [inline GVRf/Framework/backend_oculus/src/main/java/org/gearvrf/OvrXMLParser.java]: added the check to the day dream parser and verified by running the gvr-video sample on a day dream phone.

**Comment by liaxim** [comment]
Will merge Tuesday after lunch unless there are comments.

**Pull request merged**